cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/DistributedStreamService.java

```diff
@@ -369,14 +380,28 @@ private void createHeartbeatsFeed() throws NotificationFeedException {
       .setDescription("Stream heartbeats feed.")
       .build();
 
+    LOG.debug("Ensuring Stream HeartbeatsFeed exists.");
+    boolean isRetry = false;
     while (true) {
       try {
         feedManager.getFeed(streamHeartbeatsFeed);
+        LOG.debug("Stream HeartbeatsFeed exists.");
         return;
-      } catch (NotificationFeedNotFoundException e) {
+      } catch (NotificationFeedNotFoundException notFoundException) {
+        if (!isRetry) {
+          LOG.debug("Creating Stream HeartbeatsFeed.");
+        }
         feedManager.createFeed(streamHeartbeatsFeed);
+        LOG.debug("Stream HeartbeatsFeed created.");
```

Comment:
should this be info/warn ? if there was exception and on retry if creation is successful, assuming the log.level is warn, we would see the error and going to retry message but not the successful creation message.